src/nupic/types/Sdr.cpp

```diff
@@ -339,9 +339,9 @@ namespace nupic {
         }
         // Check data
         return std::equal(
-            getDense().begin(),
-            getDense().end(), 
-            sdr.getDense().begin());
+            getSparse().begin(),
```

Comment:
The sdr does not currently require that the sparse data be sorted, so this wont work.  We could add this requirement, but id rather do that in a separate PR, and i honestly don't want to force all SDRs to be sorted.

Reply:
it's good not to require sorted sparse, I'll change this back, thanks!